### PR TITLE
Container refactor

### PR DESCRIPTION
**Pull request by ehelms**
This is a large set of changes from a branch that has had active development to build out the container work and re-factor a good chunk of how this is intended to work.

### CODE REVIEW

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

File: containers/docs/deploy.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    sudo yum install ansible
+
+Using pip:
+
+    sudo pip install ansible
+
+### Setup Docker
+
+In order to build the containers, run them locally or use OpenShift, Docker needs to be installed and configured with an insecure registry.
```
Comment (ekohl) [inline containers/docs/deploy.md]: Maybe emphasize that it's needed even with a remote openshift here.

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

**Comment by ohadlevy** [comment]
@ehelms if i follow the test failure correctly, the test.rb failed but afterwards ansible was successful creating resources in foreman via the api? is that a failure of the test.rb?

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
@ohadlevy The ansible failed because it ran `test.rb` which failed on the smoke tests via the output. The current failure seems to be somewhat transient. It occurrs about half the time on Jenkins and not once locally for me.

**Comment by ehelms** [comment]
@ekohl I cleaned up the docs based on your comments. Got a green run now with some modifications.

**Comment by ehelms** [comment]
[test containers]

File: containers/deploy/foreman/deployments.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@
     - stop
     - restart
 
-- name: Create deployment, and scale replicas up
+- name: Create Foreman deployment
```
Comment (carbonin) [inline containers/deploy/foreman/deployments.yml]: WFIW these do the create as well as the delete, we should probably name them a bit better.

File: containers/roles/foreman/files/wait_on_postgres.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
     s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     tries = 0
     print ("Waiting on postgresql to start...")
-    while not postgres_is_alive and tries < 20:
+    while not postgres_is_alive and tries < 100:
```
Comment (carbonin) [inline containers/roles/foreman/files/wait_on_postgres.py]: Why is this a different value than the one in candlepin?

File: containers/deploy/foreman/deployments.yml (hunk 2 above)
Comment (ehelms) [reply containers/deploy/foreman/deployments.yml]: That's fair. This was mostly inspired by being annoyed that I couldn't tell what stage it was at cause everything had the same name.

File: containers/roles/foreman/files/wait_on_postgres.py (hunk 3 above)
Comment (ehelms) [reply containers/roles/foreman/files/wait_on_postgres.py]: Hah, fair question. I think I just kept adjusting them till things worked consistently.

**Comment by ehelms** [comment]
@ekohl Green again if you don't mind re-reviewing the docs.

My goal is still to get this in given its in a working state consistently and then iterate smaller.

File: containers/deploy/foreman/deployments.yml (hunk 2 above)
Comment (carbonin) [reply containers/deploy/foreman/deployments.yml]: Yeah, definitely understand the change you made, just seeing the diff here made me remember about the create/delete thing.

File: containers/deploy/foreman/config-maps.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+              namespace: "{{ project_name }}"
+          data:
+            application.conf: |
+              # Timeout: The number of seconds before receives and sends time out.
+              Timeout 120
+              RewriteEngine On
+              Options SymLinksIfOwnerMatch
+              Listen 443
+              <VirtualHost *:443>
+               RequestHeader set X_FORWARDED_PROTO 'https'
```
Comment (ekohl) [inline containers/deploy/foreman/config-maps.yml]: Nit: odd space missing in indenting

File: containers/deploy/foreman/config-maps.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                SSLCertificateFile      "/etc/pki/katello/certs/katello-server.crt"
+                SSLCertificateKeyFile   "/etc/pki/katello/private/katello-server.key"
+                SSLVerifyClient         optional
+                SSLCACertificateFile    "/etc/pki/katello/certs/katello-default-ca.crt"
+                SSLVerifyDepth          3
+                SSLOptions +StdEnvVars
+
+                # SSL Proxy directives
+                SSLProxyEngine On
+                SSLProxyCheckPeerCN off
```
Comment (ekohl) [inline containers/deploy/foreman/config-maps.yml]: Does this need a TODO?

**Comment by carbonin** [comment]
:tada: :sparkles: :+1:

**Pull request merged**